application_test.go

```diff
@@ -198,14 +198,33 @@ func TestApplicationEnvs(t *testing.T) {
 	app := NewDockerApplication()
 	assert.Nil(t, app.Env)
 
-	app.AddEnv("hello", "world").AddEnv("foo", "bar")
-	assert.Equal(t, 2, len(*app.Env))
+	app.AddEnv("hello", "world").AddEnv("foo", "bar").AddSecret("top", "secret1", "/path/to/secret")
+	assert.Equal(t, 2, len((*app.Env)))
```

Comment:
Please wrap this in an if statement and move the assertions on the content of `app.Env` (lines 204 and 205) inside the if block.